packages/grafana-o11y-ds-frontend/src/TemporaryAlert.tsx

```diff
@@ -0,0 +1,39 @@
+import { css } from '@emotion/css';
+import React, { useEffect } from 'react';
+
+import { Alert, AlertVariant, useTheme2 } from '@grafana/ui';
+
+export const TemporaryAlert = (props: {
+  setVisible: (visible: boolean) => void;
+  severity: AlertVariant;
+  text: string;
+  visible: boolean;
+}) => {
+  const theme = useTheme2();
+
+  useEffect(() => {
+    if (props.visible) {
+      setTimeout(() => {
+        props.setVisible(false);
+      }, 3000);
+    }
+  }, [props]);
```

Comment:
```suggestion
  useEffect(() => {
  let timer: ReturnType<typeof setTimeout>;
    if (props.visible) {
      timer = setTimeout(() => {
        props.setVisible(false);
      }, 3000);
      
      return () => {
        clearTimeout(timer)
      }
    }
  }, [props]);
```
Looking at my code, I forgot to do it, but we should clear timer. Something like code above.